src/api/app.js

```diff
@@ -293,6 +298,71 @@ module.exports.networkState = (appHandle) => {
     .then((obj) => obj.app.networkState);
 };
 
+/**
+ * Verify whether or not network state is INIT
+ * @name window.safeApp.isNetStateInit
+ *
+ * @param {SAFEAppHandle} appHandle
+ *
+ * @returns {Boolean}
+ * */
+module.exports.isNetStateInit = (appHandle) => {
+  return getObj(appHandle)
+  .then((obj) => obj.app.isNetStateInit());
+};
+
+/**
+ * Verify whether or not network state is CONNECTED
+ * @name window.safeApp.isNetStateConnected
+ *
+ * @param {SAFEAppHandle} appHandle
+ *
+ * @returns {Boolean}
+ * */
+module.exports.isNetStateConnected = (appHandle) => {
+  return getObj(appHandle)
+  .then((obj) => obj.app.isNetStateConnected());
+};
+
+/**
+ * Verify whether or not network state is DISCONNECTED
+ * @name window.safeApp.isNetStateDisconnected
+ *
+ * @param {SAFEAppHandle} appHandle
+ *
+ * @returns {Boolean}
+ * */
+module.exports.isNetStateDisconnected = (appHandle) => {
+  return getObj(appHandle)
+  .then((obj) => obj.app.isNetStateDisconnected());
+};
+
+/**
+ * Resets the object cache kept by the underlyging library
+ * @name window.safeApp.clearObjectCache
+ *
+ * @param {SAFEAppHandle} appHandle
+ *
+ * @returns {Boolean}
+ * */
+module.exports.clearObjectCache = (appHandle) => {
+  return getObj(appHandle)
+  .then((obj) => obj.app.clearObjectCache());
```

Comment:
I think for this we also need to remove all the handles we have for the app, although we need to keep the app's handle.

Reply:
All other requests have been address but I don't understand this one.